frontend/public/components/nav.jsx

```diff
@@ -234,6 +234,17 @@ const NavSection = connect(navSectionStateToProps)(
 
 const Sep = () => <div className="navigation-container__section__separator" />;
 
+const ClusterPickerNavSection = connectToFlags(FLAGS.OPENSHIFT)(({flags}) => {
+  // Hide the cluster picker on OpenShift clusters. Make sure flag detection is complete before showing the picker.
+  if (flags[FLAGS.OPENSHIFT] !== false) {
```

Comment:
Why not eliminate/encode the comment by doing:

```JavaScript 
if(isOpenShiftCluster()) {
   return null;
}
```
Or even:

```JavaScript 
return isOpenShiftCluster()
  ?  null 
  : <div className="navigation-container__section navigation-container__section--cluster-picker">
    <ClusterPicker />
 </div>;
```
I'd rather encode the logic & use a named function than use comments to describe the logic.

Reply:
Maybe I'm misunderstanding how `!== false` makes `sure flag detection is complete before showing the picker`.

Reply:
`undefined` means flag detection is not complete. If we used `!flags[FLAGS.OPENSHIFT]` for instance, the cluster picker would flicker visible then disappear on load. We only display the picker when exactly set to `false`

Reply:
If `undefined` and `false` have different meanings, we should strive to improve this. It's an antipattern & contrary to most programming in JS, where we treat `falsy` as `falsy`.

I don't want to hold up the PR over this one, esp if its a problem thoughout the codebase.